zson/table.go

```diff
@@ -41,7 +46,24 @@ func (t *TypeTable) LookupType(zson string) (zng.Type, error) {
 	defer t.mu.Unlock()
 	typ, ok := t.toType[zson]
 	if !ok {
-		panic("zson parser not yet implemented: see issue #1679")
+		zp, err := NewParser(strings.NewReader(zson))
+		if err != nil {
+			return nil, err
+		}
+		ast, err := zp.ParseValue()
+		if ast == nil || err != nil {
+			return nil, err
+		}
+		a := NewAnalyzer()
+		val, err := a.ConvertValue(t.zctx, ast)
+		if err != nil {
+			return nil, err
+		}
+		tv, ok := val.(*TypeValue)
+		if !ok {
+			return nil, fmt.Errorf("(*TypeTable).LookupType() internal error: value of type %T", val)
```

Comment:
Nit: No other error strings use "(*Type).Method" notation, and it'd be nice to be consistent.

Reply:
I can make the change, but what's the proper form - just start with `"LookupType internal error:"`?

Reply:
Based on what I see in package zson, I'd vote for something like `"internal error: value of type %T in TypeTable.LookupType()"`.
```
$ git grep -i '(errors.New|fmt.Errorf)\("[^"]*internal'
microindex/writer.go:                   return fmt.Errorf("internal microindex error: index file size (%d) does not equal zng size (%d)", n, size)
zio/zjsonio/stream.go:          return nil, errors.New("internal error: zng record body must be a container")
zio/zngio/scanner.go:                   return nil, errors.New("internal error: batch end before zngio batch started")
zson/analyzer.go:       return nil, fmt.Errorf("internal error: unknown ast type in Analyzer.convertAny(): %T", val)
zson/analyzer.go:                       return nil, fmt.Errorf("internal error built type (%q) does not match semantic type (%q)", zv.Type.ZSON(), typ.ZSON())
```

Reply:
Nicer! https://github.com/brimsec/zq/pull/1834